File: PY/gcdLcm.py
# 두 정수를 입력 받아 최대공약수와 최소공배수 구하기

def gcd(n1, n2):
    
    i = min(n1,n2)

    if n1%i == 0 and n2%i == 0:
        return i

    i-=1

def lcm(n1, n2):

   return n1 * n2/gcd(n1,n2)


def solution(n1, n2):

    answer = []

    answer.insert(0, gcd(n1,n2))
    answer.insert(1, lcm(n1,n2))

    return answer

print(solution(3, 6))

File: JS/gcdLcm.js
// 두 정수의 최대공약수, 최소공배수 구하기

function soluction(num1, num2) {
    let answer = [];

    function gcd(num1, num2) {

        let num = num1 > num2 ? num1 : num2;
        let maxNum;

        for (let i = 1; i <= num; i++) {
            if (num1 % i === 0 && num2 % i === 0) {
                maxNum = i;
            }
        }

        return maxNum;
    }

    function lcm(num1, num2) {
        return num1 * num2 / gcd(num1, num2);
    }

    answer.push(gcd(num1, num2));
    answer.push(lcm(num1, num2));

    return answer;
}

console.log(soluction(1, 6));